e2e/utils/functions/testBase.ts

```diff
@@ -434,13 +448,102 @@ export class HomePage {
+    await this.page.getByRole('row', { name: `${randomSupersetRoleName.roleName}` }).getByRole('link').nth(1).click();
+    await delay(2000);
+    await this.page.getByPlaceholder('Name').clear();
+    await this.page.getByPlaceholder('Name').fill(`${randomSupersetRoleName.updatedRoleName}`);
+    await this.page.locator('button[type="submit"]').click();
+    await delay(2000);
+    await expect(this.page.getByText('Changed Row')).toBeVisible();
+    await expect(this.page.getByText(`${randomSupersetRoleName.updatedRoleName}`)).toBeVisible();
+    await delay(30000);
+  }
+
+  async deleteSupersetRole(){
+    await this.page.goto(`${E2E_ANALYTICS_URL}/roles/list`);
+    await this.page.getByRole('row', { name: `${randomSupersetRoleName.roleName}` }).getByRole('checkbox').check();
+    await this.page.getByRole('row', { name: `${randomSupersetRoleName.roleName}` }).getByRole('link').nth(2).click();
+    await delay(2000);
+    await this.page.getByRole('link', { name: 'OK' }).click();
+    await delay(2500);
+    await expect(this.page.getByText(`Deleted Row`)).toBeVisible();
+    await expect(this.page.getByText(`${randomSupersetRoleName.roleName}`)).not.toBeVisible();
+  }
+
+  async deleteUpdatedSupersetRole(){
+    await this.page.goto(`${E2E_ANALYTICS_URL}/roles/list`);
+    await this.page.getByRole('row', { name: `${randomSupersetRoleName.updatedRoleName}` }).getByRole('checkbox').check();
+    await this.page.getByRole('row', { name: `${randomSupersetRoleName.updatedRoleName}` }).getByRole('link').nth(2).click();
+    await delay(2000);
+    await this.page.getByRole('link', { name: 'OK' }).click();
+    await delay(2500);
+    await expect(this.page.getByText(`Deleted Row`)).toBeVisible();
+    await expect(this.page.getByText(`${randomSupersetRoleName.updatedRoleName}`)).not.toBeVisible();
+  }
+
+  async deleteSyncedSupersetRoleInKeycloak() {
+    await this.page.getByRole('row', { name: `${randomSupersetRoleName.roleName}` }).getByLabel('Actions').click();
+    await this.page.getByRole('menuitem', { name: 'Delete' }).click();
+    await this.page.getByTestId('confirm').click();
+    await expect(this.page.getByText(`The role has been deleted`)).toBeVisible();
+  }
+
   async goToClients() {
     await this.page.getByTestId('realmSelectorToggle').click();
     await this.page.getByRole('menuitem', { name: 'ozone' }).click();
     await this.page.getByRole('link', { name: 'Clients' }).click();
     await delay(2000);
   }
 
+  async goToOpenMRSRole() {
+    await this.page.getByRole('link', { name: 'openmrs', exact: true }).click();
+    await this.page.getByTestId('rolesTab').click();
+  }
+
+  async goToOpenMRSRoleAttributes() {
+    await this.page.getByRole('link', { name: `${randomOpenMRSRoleName.roleName}` }).click();
+    await this.page.getByTestId('attributesTab').click();
+  }
+
+  async goToSupersetRole() {
+    await this.page.getByLabel('Pagination top').getByLabel('Go to next page').click();
```

Comment:
Is there a better way to handle pagination?
 I'd like to raise a concern regarding the default page size setting. If an individual sets the default page size to 25 or any higher value(or we add or reduce the number of clients, superset can be on any page), this means the entire clients' list is displayed on the initial page. Consequently, there will be NO "**_go to next page_**" button thus breaking this functionality. I recommend implementing a solution where you check for all pages for a given client.